server/routes/shared/appointment/feedback/submittedFeedbackPresenter.ts

```diff
@@ -20,21 +20,31 @@ export default class SubmittedFeedbackPresenter extends FeedbackAnswersPresenter
   constructor(
     appointmentDetails: ActionPlanAppointment | Appointment,
     private readonly serviceUser: DeliusServiceUser,
+    private readonly userType: 'probation-practitioner' | 'service-provider',
+    private readonly referralId: string,
+    private readonly actionPlanId: string | null = null,
     private readonly assignedCaseworker: User | null = null
   ) {
     super(appointmentDetails)
+
     if (this.isActionPlanAppointment(appointmentDetails)) {
       this.attendancePresenter = new ActionPlanPostSessionAttendanceFeedbackPresenter(
         appointmentDetails,
         this.serviceUser
       )
-      this.behaviourPresenter = new ActionPlanSessionBehaviourFeedbackPresenter(appointmentDetails, this.serviceUser)
+      this.behaviourPresenter = new ActionPlanSessionBehaviourFeedbackPresenter(
```

Comment:
I still think it would be nicer to have a component that encapsulates the `behaviour feedback` question and answers separate from other concerns (such as backlink and form presentation). I feel that passing all these null parameters becomes unwieldly.

We have a class called `feedbackAnswersPresenter.ts` which contains `AttendanceFeedbackPresenter` and `BehaviourFeedbackPresenter`.

I believe that this class doesn't need to be concerned with form details because we only ever use it after submission to the back-end has occurred. So we can remove the dependency to AttendanceFeedbackPresenter and BehaviourFeedbackPresenter and instead just use the appointment directly.
We can possibly also define the questions here as well (effectively inverting the relationship). In this way it would be more useable on other pages.

What do you think?

I realise this would be quite an involved change to do, but if you're in agreement with this - perhaps one of us can do it after this PR has been merged.

Reply:
I think I see what you mean - it does feel like we're tying a lot of things together and passing in more than we need. I feel like you have a clearer image of how this could look in your head, would you be happy to take it on after this PR is merged?

Reply:
Sure. It might not work once I get down to the details!